src/main.rs

```diff
@@ -195,6 +196,14 @@ fn run() -> i32 {
         }
     }
 
+    if matches.opt_present("p") {
+        if path == "-" {
+            eprintln!("stdout is used to print the image, -p option ignored");
```

Comment:
I would much rather make conflicting options a hard error.

Reply:
The `path` is computed pretty late; would you rather error out very late, or should I move the `ts_path` earlier?

Reply:
I'd say move it up to the top where I check how many "free" arguments were passed. If both `-p` and a path were passed, then abort.

Reply:
Actually you're right, we need to check whether we're outputting to stdout for this to make sense (though again I don't really see the point of printing a path that was explicitly passed on the command line).